man/man8/zpool.8

```diff
@@ -579,6 +579,27 @@ Keep in mind is that the \fBashift\fR is \fIvdev\fR specific and is not a \fIpoo
 
 .sp
 .LP
+The following properties can be set at creation time:
+.ne 2
+.na
+.sp
+\fB\fBsegregate_log\fR=\fIon\fR\fR
+.sp
+\fB\fBsegregate_metadata\fR=\fIon\fR\fR
+.sp
+\fB\fBsegregate_smallblks\fR=\fIon\fR\fR
+.ad
+.sp .6
+.RS 4n
+The segregate properties direct the allocating vdevs in the pool
+to segregate a portion of their metaslabs to the allocation class
+indicated by the property's name. These properties require that
```

Comment:
We should document for the naive administrator *when* they would want to turn these on or off.  To be honest I'm not sure I know when it would be appropriate to turn them on or off.  I'm a bit concerned that we are giving folks a sharp knife here.  Given that these settings are (AFAIK) entirely for improving performance, is there any way that we could make ZFS "do the right thing" in most circumstances, without need for these properties?

E.g. I bet that for ZIL, it would almost always be a win to reserve a metaslab for ZIL blocks (at least, if there isn't a whole device dedicated to the ZIL).  Maybe we could do something like that, but slightly more dynamically, for small/metadata blocks?